src/Adapter/MSTest.CoreAdapter/Execution/TestClassInfo.cs

```diff
@@ -306,40 +311,52 @@ public string RunClassCleanup()
                 return null;
             }
 
-            lock (this.testClassExecuteSyncObject)
+            if (this.IsClassInitializeExecuted)
```

Comment:
Nit: how about `this.ClassCleanupMethod == null || this.IsClassInitializeExecuted == false`?